packages/interactivity/src/utils.js

```diff
@@ -1,16 +1,23 @@
 /**
  * External dependencies
  */
-import { useRef, useEffect } from 'preact/hooks';
+import { useEffect } from 'preact/hooks';
 import { effect } from '@preact/signals';
 
-function afterNextFrame( callback ) {
-	const done = () => {
-		window.cancelAnimationFrame( raf );
-		setTimeout( callback );
-	};
-	const raf = window.requestAnimationFrame( done );
-}
+const afterNextFrame = ( callback ) => {
+	return new Promise( ( resolve ) => {
+		const done = () => {
+			clearTimeout( timeout );
+			window.cancelAnimationFrame( raf );
+			setTimeout( () => {
+				callback();
+				resolve();
+			} );
+		};
+		const timeout = setTimeout( done, 100 );
+		const raf = window.requestAnimationFrame( done );
```

Comment:
Why not simplify this part instead of calling `setTimeout` so many times?

I understand there should be a reason, as the Preact implementation was also doing it, right? 😅 

```suggestion
		window.requestAnimationFrame( () => {
			callback();
			resolve();
		} );
```

The code above passes all tests (at least locally). ☝️

Reply:
It's the [exact same logic used by default on `useEffect`](https://github.com/preactjs/preact/blob/main/hooks/src/index.js#L431-L453), so I think we should keep it as it is to keep the behavior consistent.

According to Preact's code comments, the first `setTimeout` is added to do it _after_ the next frame, not before:

> Do this by combining requestAnimationFrame (rAF) + setTimeout to invoke a callback after the next browser frame.

And the second `setTimeout` is to make sure that the callback is invoked if RAF is not fired:

> Also, schedule a timeout in parallel to the the rAF to ensure the callback is invoked even if RAF doesn't fire (for example if the browser tab is not visible)

Reply:
Many thanks for the explanation, Luis. 💯